src/BaseList/BaseList.tsx

```diff
@@ -0,0 +1,153 @@
+import clsx from "clsx";
+import React from "react";
+import {
+  ListContext,
+  ListGridContext,
+  useGridStyles,
+  useListContext,
+} from "./context";
+import { ListGridTemplate, useStyles } from "./styles";
+
+export interface BaseListProps extends React.HTMLAttributes<HTMLElement> {
+  gridTemplate: ListGridTemplate;
+}
+export const BaseList: React.FC<BaseListProps> = ({
+  children,
+  gridTemplate,
+  ...props
+}) => (
+  <ListGridContext.Provider value={gridTemplate}>
+    <section {...props}>{children}</section>
+  </ListGridContext.Provider>
+);
+
+export type BaseListItemClassKey =
+  | "row"
+  | "rowBody"
+  | "rowHead"
+  | "rowFoot"
+  | "rowHover"
+  | "rowBodySelected";
+
+export interface BaseListItemProps
+  extends React.HTMLAttributes<HTMLDivElement | HTMLLIElement> {
+  classes: Record<BaseListItemClassKey, string>;
+  hover?: boolean;
+  selected?: boolean;
+}
+
+export const BaseListItem: React.FC<BaseListItemProps> = ({
+  classes,
+  className,
+  children,
+  hover = true,
```

Comment:
Can we name this prop more clearly?

Reply:
I don't see how this prop can be misunderstood or misused. Also, guys at Material-UI didn't come up with anything clearer:
https://mui.com/api/table-row/#main-content

If this still bothers you I can add docstring to field declaration.